djangocms_versioning/helpers.py

```diff
@@ -43,3 +44,35 @@ def replace_admin_for_models(models, admin_site=None):
         except KeyError:
             continue
         _replace_admin_for_model(modeladmin, admin_site)
+
+
+def register_versionadmin_proxy(versionable, admin_site=None):
+    """Creates a model admin class based on `VersionAdmin` and registers
+    it with `admin_site` for `versionable.version_model_proxy`.
+
+    This model class applies filtering on the list of versions,
+    so that only versions for `versionable.content_model` are shown.
+
+    :param versionable: VersionableItem instance
+    :param admin_site: AdminSite instance
+    """
+    if admin_site is None:
+        admin_site = admin.site
+
+    if versionable.version_model_proxy in admin_site._registry:
+        # Attempting to register the proxy again is a no-op.
+        return
```

Comment:
There's still no warning logged here (or exception thrown - depending on how we think it's better to handle). I do think doing this silently isn't too great.